lib/core/storages/memory.ts

```diff
@@ -24,8 +23,57 @@ export class MemoryStorage implements IStorage {
     tags: Record<string, string> | undefined,
     limit: number | undefined,
   ): AsyncGenerator<DurablePromise[], void> {
-    // for now WithTimeout will implement
-    // search logic
     yield Object.values(this.promises);
   }
 }
+
+export class MemoryScheduleStorage implements IScheduleStorage {
+  private schedules: Record<string, Schedule> = {};
+
+  constructor() {}
+
+  async rmw<P extends Schedule | undefined>(id: string, f: (item: Schedule | undefined) => P): Promise<P> {
+    const item = f(this.schedules[id]);
+    if (item) {
+      this.schedules[id] = item as Schedule;
+    }
+
+    return item;
+  }
+
+  async *search(
+    id: string,
+    state: string | undefined,
+    tags: Record<string, string> | undefined,
+    limit: number | undefined,
+  ): AsyncGenerator<Schedule[], void> {
+    yield Object.values(this.schedules);
+  }
+
+  async deleteSchedule(id: string): Promise<boolean> {
+    try {
+      delete this.schedules[id];
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+}
+
+export class MemoryStorage {
```

Comment:
See [this comment](https://github.com/resonatehq/resonate-sdk-ts/pull/21/files#r1457783031), once we instantiate the LocalStore with it's constituent parts we should be able to get rid of the MemoryStorage.